redash_toolbelt/examples/find-table-names.py

```diff
@@ -0,0 +1,92 @@
+import itertools, json, re
+import click
+from redash_toolbelt import Redash
+
+
+# This regex captures three groups:
+#
+#   0. A FROM or JOIN statement
+#   1. The whitespace character between FROM/JOIN and table name
+#   2. The table name
+PATTERN = re.compile(r"(FROM|JOIN)( )([^\s\(\)]+)", flags=re.IGNORECASE)
```

Comment:
In addition:
- there may be more than one whitespace between FROM and table name, and it may be any whitespace character (e.g. tab). `\s+` should be used instead; also, it's good idea to use `re.UNICODE` flag;
- there is another syntax: `SELECT ... FROM table_a, table_b, ..., table_N` (IIRC it works as INNER JOIN) - this regex will capture first table with comma and won't capture the rest.

Reply:
Smart! Fixed in fccb4bd

Reply:
> Smart! Fixed in fccb4bd

I guess you need to update comment above regex as well :smile:

Reply:
Apparently extracting table names can get tricky. Consider using `sqlparse` for this? (check [this example](https://github.com/andialbrecht/sqlparse/blob/master/examples/extract_table_names.py) out)

Reply:
I completely missed @kravets-levko 's feedback because my page didn't update. Looking into this now.

Reply:
I tried `sqlparse` initially when writing this but was unimpressed:

+ It's really _slow_
+ It's finicky and doesn't return results where I expect it to

I tried subbing it into the current logic and it doesn't even find tables the way this regex does. It misses the `events` table, for example.

I should add some tests.

Reply:
Will revisit this in the morning. I pushed 0a8786c which includes tests.

Will need a slightly different approach for the comma-separated INNER JOINs @kravets-levko mentioned.

Reply:
![image](https://user-images.githubusercontent.com/289488/79162487-0b138480-7de6-11ea-8c4f-2f9be383b12f.png)

Reply:
We use sqlparse in Redash to format the queries. It might have some issues, but I would be surprised if it has less detections than the regex. 

But the regex is good enough for now, let's table this and follow up in the future.

Reply:
Puns shall not be tolerated!

Reply:
I handled @kravets-levko's situation with comma-delimitted JOINs by formatting the query before the `regex.findall()` call. If the query uses the fancy INNER JOIN syntax of `table1,table2,table3`, all three are now captured.

Aliased table names are covered in `test_2`.